cli/azd/pkg/state/config.go

```diff
@@ -0,0 +1,12 @@
+package state
+
+// Config is the state configuration for an azd project
+type Config struct {
```

Comment:
Not a huge fan of splaying the `azd` configuration schema across multiple packages. I would prefer if the configuration lives in one root package for the config file, and we can certainly create new structs for domain-specific handling if needed.